dll/win32/shimgvw/shimgvw.c

```diff
@@ -959,6 +977,58 @@ ImageView_OnSize(HWND hwnd, UINT state, INT cx, INT cy)
     }
 }
 
+static LRESULT CALLBACK
+ImageView_Delete(HWND hwnd)
+{
+    DPRINT1("ImageView_Delete: unimplemented.\n");
+    return 0;
+}
+
+static LRESULT CALLBACK
+ImageView_Modify(HWND hwnd)
+{
+    int nChars = GetFullPathNameW(currentFile->FileName, 0, NULL, NULL);
+    WCHAR *lpFPN;
+    SHELLEXECUTEINFOW sei;
+
+    if (!nChars)
+    {
+        DPRINT1("ImageView_Modify: failed to get full path name.\n");
+        return 1;
+    }
+
+    lpFPN = (WCHAR *)malloc(nChars * sizeof(WCHAR));
```

Comment:
```suggestion
    pszPathName = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, nChars * sizeof(WCHAR));
```

Reply:
The whole source is using malloc() so, when I had written this patch, I prefered to use that instead of HeapAlloc(). And why HEAP_ZERO_MEMORY? The buffer is allocated of the exact size we need, from the information we got from previous call to GetFullPathNameW(). There is absolutely no need to do so.

Reply:
HEAP_ZERO_MEMORY is unnecessary. Sorry.